pkg/kube/kube_nodes.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kube
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/ethersphere/node-funder/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"log"
+	"math/big"
+	"net/http"
+	"os"
+	"sync"
+)
+
+const (
+	beeAddressEndpoint = "/addresses"
+	beeWalletEndpoint  = "/wallet"
+)
+
+type NamespaceNodes struct {
+	Name  string
+	Nodes []Node
+}
+type Node struct {
+	Name      string
+	Ip        string
+	BeeTokens BeeTokens
+}
+
+type BeeTokens struct {
+	EthAddress      string
+	ChainID         int
+	ContractAddress string
+	ETH             *big.Int
+	BZZ             *big.Int
+	GBZZ            *big.Int
+}
+
+func NewKube() (*corev1client.CoreV1Client, error) {
+	config, err := GetConfig()
+	if err != nil {
+		log.Printf("get configuration failed with error: %v", err)
+		return nil, err
+	}
+
+	// Create a Kubernetes core/v1 client.
+	coreClient, err := corev1client.NewForConfig(config)
+	if err != nil {
+		log.Printf("creating Kubernetes client failed with error: %v", err)
+		return nil, err
+	}
+	return coreClient, nil
+}
+
+func GetConfig() (*rest.Config, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		log.Printf("obtaining user's home dir: %v", err)
+		return nil, err
+	}
+	kubeconfigPath := home + "/.kube/config"
+	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfigPath},
+		&clientcmd.ConfigOverrides{ClusterInfo: api.Cluster{Server: ""}}).ClientConfig()
+}
+
+func GetNodeInfo(ctx context.Context, kube *corev1client.CoreV1Client, namespace string) (*NamespaceNodes, error) {
+	// List all Pods in our current Namespace.
+	pods, err := kube.Pods(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Printf("listing pod failed with error: %v", err)
+		return nil, err
+	}
+
+	nodes := make([]Node, 0)
+	var wg sync.WaitGroup
+	for _, pod := range pods.Items {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			beeTokens, err := GetTokens(ctx, pod.Status.PodIP)
```

Comment:
Preferably we should use channels as means of communication between goroutines.

Reply:
So channel implementation might not be possible because we are not sure about the number of pods. Sometimes it also fetches cron jobs, which do not require funding. 

So if the pods are 20 and jobs are 2. Pods will iterate 22 times.